### PR TITLE
Add support for the new table pagination layout

### PR DESCRIPTION
**Pull request by kharbutli-meta**
## Description

GTT-1919: Adds a new table pagination layout and switches all the application's tables (table widget, dashboard listing, user listing ..etc) to it.

Screen Shot 2022-04-26 at 12 23 42 PM

## Testing

* Unit tests - updated and added.
* Local instance verification

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: frontend/src/components/Pagination.scss [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ }
+
+ .dots {
+  color: #005ea2;
+ }
+
+ .paginationItem {
+  height: 40px;
+  width: 40px;
+  color: #005ea2;
```
Comment (mabreuortega) [inline frontend/src/components/Pagination.scss]: This color needs to be updated. We are not using any blue in the theme so far.
For reference use this:
https://designsystem.digital.gov/utilities/color/

File: frontend/src/components/Pagination.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  background: #fff;
```
Comment (mabreuortega) [inline frontend/src/components/Pagination.scss]: use theme colors

```sass
# top of the file
import "../../base.scss";

.paginationItem {
   ...
   border: color("while");
   ...
}
```

File: frontend/src/components/Pagination.scss [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  vertical-align: middle;
+  line-height: 40px;
+  position: relative;
+  margin: 5px;
+  cursor: pointer;
+ }
+ 
+ .paginationItem.current {
+  color: #fff;
+  background: #000;
```
Comment (mabreuortega) [inline frontend/src/components/Pagination.scss]: this should be using 

```sass
color("base-darker")
```

File: frontend/src/components/Pagination.scss [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  text-align: center;
+  vertical-align: middle;
+  line-height: 40px;
+  position: relative;
+  margin: 5px;
+  cursor: pointer;
+ }
+ 
+ .paginationItem.current {
+  color: #fff;
```
Comment (mabreuortega) [inline frontend/src/components/Pagination.scss]:
```sass
color("white")
```

File: frontend/src/components/Pagination.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+          </div>
+        ) : (
+          <div className="dots" key={`dots ${page}`}>
+            &thinsp; &#8230; &thinsp;
+          </div>
+        );
+      })}
+
+      {props.currentPage < props.numPages && (
+        <div
```
Comment (mabreuortega) [inline frontend/src/components/Pagination.tsx]: I really feel we should be using the Button component here and adjust css.

**Review by mabreuortega — CHANGES_REQUESTED** [review]
Overall looks great but we need to work on the mobile view

image

This is not passing the 400% zoom in test. Also, please workout heh first and last icon/text split when reducing the screen size.

Screen Shot 2022-04-26 at 2 44 29 PM

File: frontend/src/components/Pagination.tsx (hunk 5 above)
Comment (kharbutli-meta) [reply frontend/src/components/Pagination.tsx]: Switched to the button component

File: frontend/src/components/Pagination.scss (hunk 1 above)
Comment (kharbutli-meta) [reply frontend/src/components/Pagination.scss]: Fixed

Comment (kharbutli-meta) [reply frontend/src/components/Pagination.scss]: Fixed

File: frontend/src/components/Pagination.scss (hunk 2 above)
Comment (kharbutli-meta) [reply frontend/src/components/Pagination.scss]: Fixed

File: frontend/src/components/Pagination.scss (hunk 4 above)
Comment (kharbutli-meta) [reply frontend/src/components/Pagination.scss]: Fixed

File: frontend/src/components/Pagination.scss (hunk 3 above)
Comment (kharbutli-meta) [reply frontend/src/components/Pagination.scss]: Fixed

**Review by mabreuortega — APPROVED** [review]
LGTM. great refactor!

**Pull request merged**